xbmc/windowing/WinSystem.h

```diff
@@ -120,6 +120,19 @@ class CWinSystemBase
   virtual bool IsTextInputEnabled() { return false; }
 
   std::string GetClipboardText(void);
+  
+  /**
+   * Get VADisplay that can be used for hardware decoding
+   * 
+   * Kodi currently does not use libva backend-specific interfaces for presenting
+   * the decoded frames.
+   * If this changes, the API for interacting with the windowing system
+   * will have to be augmented.
+   * 
+   * \return new VADisplay that can be used for hardware decoding or nullptr
+   *         if unsupported or creation failed
+   */
+  virtual void* GetVaDisplay() { return nullptr; }
```

Comment:
As you know, one (admittedly lower priority) goal of this summer of code project is to have both X11 and Wayland support in one Kodi build (#11). Or put more generally, multiple windowing implementations. This means that most things will have to be virtual then because  `g_Windowing` will be `CWinSystemBase` and not the concrete implementation as it is now as you do not know which it is going to be at compile time. I'm just preparing for that. Can remove the `virtual` and add it when that change eventually happens though. Maybe there will be a different solution for this altogether, for example a `IVAAPIDisplayProvider` interface or something. Then it would not clutter `CWinSystemBase`.